packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -94,17 +94,16 @@ class CurlyComponentManager {
     validatePositionalParameters(args.named, args.positional.values, definition.ComponentClass.positionalParams);
 
     if (definition.args) {
-      // args (aka 'oldArgs') may be undefined or simply be empty args, so
-      // we need to fall back to an empty array or object.
-      let newNamed = args && args.named && args.named.map || {};
-      let newPositional = args && args.positional && args.positional.values || [];
+      let newNamed = (args.named || EvaluatedNamedArgs.empty()).map;
+      let newPositional = (args.positional || EvaluatedPositionalArgs.empty()).values;
```

Comment:
Does anything fail (locally) if you remove the `|| X.empty()` guard? I am curious when that could be null